### PR TITLE
Create publisher to split output by timestamp

### PR DESCRIPTION
**Pull request by jack-moseley**
@ydai1124 please review

### CODE REVIEW

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class TimestampDataPublisher extends BaseDataPublisher {
+
+  public TimestampDataPublisher(State state) throws IOException {
+    super(state);
+  }
+
+  @Override
+  protected Path getPublisherOutputDir(WorkUnitState workUnitState, int branchId) {
+    Path outputDir = WriterUtils.getDataPublisherFinalDir(workUnitState, this.numBranches, branchId);
+    outputDir = outputDir.suffix(String.valueOf(System.currentTimeMillis()));
```
Comment (ydai1124) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: Can you add one log message for the final `outputDir`?

File: gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class TimestampDataPublisher extends BaseDataPublisher {
+
+  public TimestampDataPublisher(State state) throws IOException {
+    super(state);
+  }
+
+  @Override
+  protected Path getPublisherOutputDir(WorkUnitState workUnitState, int branchId) {
+    Path outputDir = WriterUtils.getDataPublisherFinalDir(workUnitState, this.numBranches, branchId);
```
Comment (ydai1124) [inline gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java]: You can call ` Path outputDir = super.getPublisherOutputDir(workUnitState, branchId);`, so it is more clear.

**Comment by chavdar** [comment]
Different work units for the same topic may get published to different directories. That is incorrect.

**Comment by ydai1124** [comment]
LGTM. Please squash your commits into one before merging this PR.

**Pull request merged**